packages/worker/src/nexus.ts

```diff
@@ -0,0 +1,433 @@
+    details: Buffer.from(JSON.stringify(fixBuffers(pbj))),
+  };
+}
+
+export async function handlerErrorToProto(
+  dataConverter: LoadedDataConverter,
+  err: nexus.HandlerError
+): Promise<temporal.api.nexus.v1.IHandlerError> {
+  let retryBehavior: temporal.api.enums.v1.NexusHandlerErrorRetryBehavior =
+    temporal.api.enums.v1.NexusHandlerErrorRetryBehavior.NEXUS_HANDLER_ERROR_RETRY_BEHAVIOR_UNSPECIFIED;
+  if (err.retryable === true) {
+    retryBehavior = temporal.api.enums.v1.NexusHandlerErrorRetryBehavior.NEXUS_HANDLER_ERROR_RETRY_BEHAVIOR_RETRYABLE;
+  } else if (err.retryable === false) {
+    retryBehavior =
+      temporal.api.enums.v1.NexusHandlerErrorRetryBehavior.NEXUS_HANDLER_ERROR_RETRY_BEHAVIOR_NON_RETRYABLE;
+  }
+  let { cause } = err;
+  if (cause == null) {
+    // TODO: this messes up the call stack and creates unnecessary nesting.
+    //
+    // Create an error without capturing a stack trace.
+    const wrapped = Object.create(Error.prototype);
+    wrapped.message = err.message;
+    wrapped.stack = err.stack;
+    cause = wrapped;
+  }
+  return {
+    errorType: err.type,
+    failure: await errorToNexusFailure(dataConverter, cause),
+    retryBehavior,
+  };
+}
+
+export class NexusHandler {
+  constructor(
+    public readonly taskToken: Uint8Array,
+    public readonly info: nexus.HandlerInfo,
+    public readonly abortController: AbortController,
+    public readonly handler:
+      | nexus.OperationHandler<unknown, unknown>
+      | nexus.SyncOperationHandler<unknown, unknown>
+      | undefined,
+    public readonly dataConverter: LoadedDataConverter,
+    /**
+     * Logger bound to `sdkComponent: worker`, with metadata from this Nexus task.
+     * This is the logger to use for all log messages emitted by the Nexus
+     * worker. Note this is not exactly the same thing as the Nexus context
+     * logger, which is bound to `sdkComponent: nexus`.
+     */
+    private readonly workerLogger: Logger
+  ) {
+    this.workerLogger = withMetadata(workerLogger, () => this.getLogAttributes());
+  }
+
+  protected getLogAttributes(): Record<string, unknown> {
+    return nexusLogAttributes(this.info);
+  }
+
+  protected async operationErrorToProto(
+    err: nexus.OperationError
+  ): Promise<temporal.api.nexus.v1.IUnsuccessfulOperationError> {
+    let { cause } = err;
+    if (cause == null) {
+      // Create an error without capturing a stack trace.
+      const wrapped = Object.create(Error.prototype);
+      wrapped.message = err.message;
+      wrapped.stack = err.stack;
+      cause = wrapped;
+    }
+    return {
+      operationState: err.state,
+      failure: await errorToNexusFailure(this.dataConverter, cause),
+    };
+  }
+
+  protected async startOperation(
+    payload: Payload | undefined,
+    options: nexus.StartOperationOptions
+  ): Promise<coresdk.nexus.INexusTaskCompletion> {
+    try {
+      const input = await decodeFromPayload(this.dataConverter, payload);
```

Comment:
If the data converter fails here, I assume it should result in a `BAD_REQUEST` 400 Nexus HTTP response, right? If that's correct can you confirm that it does?

Reply:
It's a test worth adding. Thanks!